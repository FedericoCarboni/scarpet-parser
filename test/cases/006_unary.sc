// Unary Operator precedence test
!id:attr && true