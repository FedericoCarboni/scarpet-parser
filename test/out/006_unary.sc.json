{
  "errors": [],
  "warnings": [],
  "root": {
    "kind": "BinaryExpression",
    "start": {
      "pos": 34,
      "row": 0,
      "col": 0
    },
    "end": {
      "pos": 50,
      "row": 0,
      "col": 0
    },
    "operator": "&&",
    "lvalue": {
      "kind": "UnaryExpression",
      "start": {
        "pos": 34,
        "row": 0,
        "col": 0
      },
      "end": {
        "pos": 43,
        "row": 0,
        "col": 0
      },
      "operator": "!",
      "value": {
        "kind": "BinaryExpression",
        "start": {
          "pos": 35,
          "row": 0,
          "col": 0
        },
        "end": {
          "pos": 43,
          "row": 0,
          "col": 0
        },
        "operator": ":",
        "lvalue": {
          "kind": "Variable",
          "start": {
            "pos": 35,
            "row": 0,
            "col": 0
          },
          "end": {
            "pos": 37,
            "row": 0,
            "col": 0
          },
          "name": "id"
        },
        "rvalue": {
          "kind": "Variable",
          "start": {
            "pos": 38,
            "row": 0,
            "col": 0
          },
          "end": {
            "pos": 43,
            "row": 0,
            "col": 0
          },
          "name": "attr"
        }
      }
    },
    "rvalue": {
      "kind": "Constant",
      "start": {
        "pos": 46,
        "row": 0,
        "col": 0
      },
      "end": {
        "pos": 50,
        "row": 0,
        "col": 0
      },
      "name": "true"
    }
  }
}