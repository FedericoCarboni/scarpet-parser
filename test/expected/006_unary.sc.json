{
    "errors": [],
    "warnings": [],
    "root": {
        "kind": "BinaryExpression",
        "start": {
            "pos": 34,
            "row": 1,
            "col": 0
        },
        "end": {
            "pos": 50,
            "row": 1,
            "col": 16
        },
        "operator": "&&",
        "lvalue": {
            "kind": "UnaryExpression",
            "start": {
                "pos": 34,
                "row": 1,
                "col": 0
            },
            "end": {
                "pos": 43,
                "row": 1,
                "col": 9
            },
            "operator": "!",
            "value": {
                "kind": "BinaryExpression",
                "start": {
                    "pos": 35,
                    "row": 1,
                    "col": 1
                },
                "end": {
                    "pos": 43,
                    "row": 1,
                    "col": 9
                },
                "operator": ":",
                "lvalue": {
                    "kind": "Variable",
                    "start": {
                        "pos": 35,
                        "row": 1,
                        "col": 1
                    },
                    "end": {
                        "pos": 37,
                        "row": 1,
                        "col": 3
                    },
                    "name": "id"
                },
                "rvalue": {
                    "kind": "Variable",
                    "start": {
                        "pos": 38,
                        "row": 1,
                        "col": 4
                    },
                    "end": {
                        "pos": 43,
                        "row": 1,
                        "col": 9
                    },
                    "name": "attr"
                },
                "comment": "// Unary Operator precedence test\n"
            }
        },
        "rvalue": {
            "kind": "Constant",
            "start": {
                "pos": 46,
                "row": 1,
                "col": 12
            },
            "end": {
                "pos": 50,
                "row": 1,
                "col": 16
            },
            "name": "true"
        },
        "comment": "// Unary Operator precedence test\n"
    }
}